src/openrct2/object/SmallSceneryObject.cpp

```diff
@@ -215,22 +206,43 @@ void SmallSceneryObject::PerformFixes()
         String::Equals(identifier, "TTRFWD07") ||
         String::Equals(identifier, "TTRFWD08"))
     {
-        static const rct_object_entry * scgMine = object_list_find_by_name("SCGMINE ");
-        if (scgMine != nullptr)
-        {
-            SetPrimarySceneryGroup((rct_object_entry *)scgMine);
-        }
+        static const rct_object_entry scgMine = GetScgMineHeader();
+        SetPrimarySceneryGroup(&scgMine);
     }
 
     // ToonTowner's glass roofs. Make them show up in the Abstract Theming.
     if (String::Equals(identifier, "TTRFGL01") ||
         String::Equals(identifier, "TTRFGL02") ||
         String::Equals(identifier, "TTRFGL03"))
     {
-        static const rct_object_entry * scgAbstr = object_list_find_by_name("SCGABSTR");
-        if (scgAbstr != nullptr)
-        {
-            SetPrimarySceneryGroup((rct_object_entry *)scgAbstr);
-        }
+        static const rct_object_entry scgAbstr = GetScgAbstrHeader();
+        SetPrimarySceneryGroup(&scgAbstr);
     }
 }
+
+rct_object_entry SmallSceneryObject::GetScgPiratHeader()
+{
+    static rct_object_entry scgPirat = { 0 };
+    scgPirat.flags = 207140231;
+    Memory::Copy(scgPirat.name, "SCGPIRAT", 8);
+    scgPirat.checksum = 21592567;
+    return scgPirat;
+}
+
+rct_object_entry SmallSceneryObject::GetScgMineHeader()
+{
+    static rct_object_entry scgMine = { 0 };
+    scgMine.flags = 207140231;
+    Memory::Copy(scgMine.name, "SCGMINE ", 8);
+    scgMine.checksum = 3638141733;
+    return scgMine;
+}
+
+rct_object_entry SmallSceneryObject::GetScgAbstrHeader()
+{
+    static rct_object_entry scgAbstr = { 0 };
+    scgAbstr.flags = 207140231;
+    Memory::Copy(scgAbstr.name, "SCGABSTR", 8);
+    scgAbstr.checksum = 932253451;
+    return scgAbstr;
+}
```

Comment:
I think these should all just call a common signature: `GetHeader(0x0C58B587, "SCGPIRAT", 0x014979F7)`

Reply:
> Also is it necessary to check flags and checksum? Why not just fix it if the name alone matches?

It goes deep into the bowels, but end up at an unordered map that uses the entire header.

With RCT2 and OpenRCT2 identifying unique objects based on flags, name and checksum, I don't want to intrude too much on this.

I can create a `CreateHeader` function, though.